web/vital_records/models.py

```diff
@@ -239,10 +236,18 @@ def get_finished():
     @transition(field=status, source="initialized", target="started")
     def complete_start(self):
         self.started_at = timezone.now()
+        return Routes.app_route(Routes.request_type)
+
+    @transition(field=status, target="type_completed")
+    def complete_type(self):
         return Routes.app_route(Routes.request_statement)
 
-    @transition(field=status, source="started", target="statement_completed")
-    def complete_statement(self):
+    @transition(field=status, target="statement_completed")
+    def complete_statement(self, request_type):
+        if request_type == "birth":
+            return Routes.app_route(Routes.request_name)
+        elif request_type == "marriage":
+            return Routes.app_route(Routes.request_name)
         return Routes.app_route(Routes.request_name)
```

Comment:
Currently, all options to go `request_name`. But when #286 is worked on, the ticket can pick up from here and specify a Marriage-specific name route on line 249.